Program.cs

```diff
@@ -1,7 +1,175 @@
+      while (true) {
+         var key = Console.ReadKey (true).KeyChar;
+         if (key == 'u' || key == 'a') return key;
+         Console.WriteLine ("Invalid input. Please select (u)nique or (a)ll solutions.");
+      }
+   }
+
+   /// <summary>Checks if placing a queen at a given position is safe</summary>
+   /// <param name="chessBoard">The current state of the chess board</param>
+   /// <param name="row">The current row to place the queen.</param>
+   /// <param name="col">The column to place the queen</param>
+   /// <param name="queenType">The type of queen ('o' for ordinary, 's' for super).</param>
+   /// <returns>True if it is safe to place the queen, false otherwise</returns>
+   static bool IsQueenSafe (int[] chessBoard, int row, int col, char queenType) {
+      for (int prevRow = 0; prevRow < row; prevRow++) {
+         int prevCol = chessBoard[prevRow];
+         if (prevCol == col || Math.Abs (prevCol - col) == Math.Abs (prevRow - row)) return false;
+         if (queenType == 's') {
+            int dx = Math.Abs (prevRow - row);
+            int dy = Math.Abs (prevCol - col);
+            if (dx == 1 && dy == 2 || dx == 2 && dy == 1) return false;
+         }
+      }
+      return true;
+   }
+
+   /// <summary>Solves the N-Queens problem using backtracking</summary>
+   /// <param name="chessBoard">The current state of the chess board</param>
+   /// <param name="slns">A list to store solutions.</param>
+   /// <param name="queenType">The type of queen ('o' for ordinary, 's' for super)</param>
+   /// <param name="slnType">The type of solutions to display ('u' for unique, 'a' for all)</param>
+   /// <param name="rows">The current row to place queens.</param>
+   /// <returns>True if a solution is found, false otherwise.</returns>
+   static bool NQueenSolver (int[] chessBoard, List<int[]> slns, char queenType, char slnType, int rows) {
+      if (rows == chessBoard.Length) {
+         PrintSolution (chessBoard, slns, slnType);
+         return true;
+      }
+      bool found = false;
+      for (int i = 0; i < chessBoard.Length; i++) {
+         if (IsQueenSafe (chessBoard, rows, i, queenType)) {
+            chessBoard[rows] = i;
+            found = NQueenSolver (chessBoard, slns, queenType, slnType, rows + 1) || found;
+         }
+      }
+      return found;
+   }
+
+   /// <summary>Prints a chessboard with queen placements for a solution</summary>
+   /// <param name="chessBoard">The current state of the chess board</param>
+   /// <param name="slns">A list of solutions</param>
+   /// <param name="slnType">The type of solutions to display ('u' for unique, 'a' for all)</param>
+   /// <param name="soln">The solution number</param>
+   static void PrintSolution (int[] chessBoard, List<int[]> slns, char slnType) {
+      if (slnType == 'a') {
+         DisplayBoard (chessBoard, slns, chessBoard.Length, slns.Count + 1);
+      } else {
+         bool isIdentical = false;

```

Comment:
```suggestion
         foreach (var s in slns)
              if (AreSolutionsIdentical (chessBoard, s)) return;
         DisplayBoard (chessBoard, slns, chessBoard.Length, slns.Count + 1);

```
Isn't this enough in the else part?